src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -298,6 +298,10 @@ def is_resources_target(target):
   def has_provides(cls, target):
     return cls.is_python_target(target) and target.provides
 
+  @classmethod
```

Comment:
Is it necessary to modify the old task? I would rather not make changes to about-to-be-deprecated code, as it adds maintenance burden. Can we make availability of this product contingent on switching to the new pipeline?